alerting/src/main/kotlin/org/opensearch/alerting/MonitorRunner.kt

```diff
@@ -757,39 +760,133 @@ object MonitorRunner : JobRunner, CoroutineScope, AbstractLifecycleComponent() {
             return
         }
 
-        for (query in queries) {
-            runForEachQuery(monitor, lastRunContext, index, query)
+        val count: Int = lastRunContext["shards_count"] as Int
```

Comment:
I created this [github issue](https://github.com/opensearch-project/alerting/issues/359) regarding this. Also this only happens when there is a reindexing operation, so there might still be issues during executions since the sequence numbers will be changed as well.